### PR TITLE
Update mbedtls configuration

### PR DESCRIPTION
**Pull request by artokin**
Define MBEDTLS_SSL_MAX_CONTENT_LEN to be 4096 bytes to avoid usage
of default value of 16k. Allocation of two mbedtls 16k buffers could
easily fail on constrained devices.

### CODE REVIEW

**Comment by artokin** [comment]
@terhei , @mikter  would you please review?

**Comment by juhhei01** [comment]
@artokin how about 6lowpan border router mbedTLS configuration? It is still use deafult one?

File: source/mbedtls_thread_config.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #define MBEDTLS_SSL_DTLS_HELLO_VERIFY
 #define MBEDTLS_SSL_EXPORT_KEYS
 #define MBEDTLS_SSL_ALL_ALERT_MESSAGES
+#define MBEDTLS_SSL_MAX_CONTENT_LEN 4096
```
[the remark this answers is not in the document]
Comment (artokin) [reply source/mbedtls_thread_config.h]: Commissioning did no work with 1500 bytes. We try to match this with MBEDTLS_SSL_MAX_FRAGMENT_LENGTH (max 4096). With this value commissioning work OK.

**Comment by artokin** [comment]
@juhhei01 , It looks like 6lowpan is not using MBEDTLS_USER_CONFIG_FILE at all. However, this change is needed for Thread SLIP configuration as well.

File: source/mbedtls_thread_config.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (juhhei01) [reply source/mbedtls_thread_config.h]: is it now allocating 8k (4k RX & 4K for tx)

**Pull request merged**